app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -380,29 +375,51 @@ def reconfigure_handle_cancel_button
           end
         end
 
+        # parameter to accept for vm reconfiguration based on role permissions
+        def reconfigure_param_list
+          list = []
+
+          if role_allows?(:feature => 'vm_reconfigure_disks')
+            list += [
+              %i[vmAddDisks disk_add],
+              %i[vmResizeDisks disk_resize],
+              %i[vmRemoveDisks disk_remove]
+            ]
+          end
+
+          if role_allows?(:feature => 'vm_reconfigure_networks')
+            list += [
+              %i[vmAddNetworkAdapters network_adapter_add],
+              %i[vmRemoveNetworkAdapters network_adapter_remove],
+              %i[vmEditNetworkAdapters network_adapter_edit],
+            ]
+          end
+
+          if role_allows?(:feature => 'vm_reconfigure_drives')
+            list += [
+              %i[vmConnectCDRoms cdrom_connect],
+              %i[vmDisconnectCDRoms cdrom_disconnect]
+            ]
+          end
+
+          list
+        end
+
         def reconfigure_handle_submit_button
           options = {:src_ids => params[:objectIds]}
           if params[:cb_memory] == 'true'
             options[:vm_memory] = params[:memory_type] == "MB" ? params[:memory] : params[:memory].to_i * 1024
           end
 
-          if params[:cb_cpu] == 'true'
+          if params[:cb_cpu] == 'true' && role_allows?(:feature => 'vm_reconfigure_memory')
```

Comment:
Why are you checking `vm_reconfigure_memory` here?

Reply:
that's a bug

Reply:
I think you should be checking `vm_reconfigure_cpu` here.